webapp/tests/karma/ts/modules/messages/messages.component.spec.ts

```diff
@@ -174,11 +201,107 @@ describe('Messages Component', () => {
     });
   });
 
-  it('ngOnDestroy() should unsubscribe from observables', () => {
-    const spySubscriptionsUnsubscribe = sinon.spy(component.subscriptions, 'unsubscribe');
+  describe('Messages breadcrumbs', () => {
+    const bettyOfflineUserContactDoc = {
+      _id: 'user',
+      parent: {
+        _id: 'parent',
+        name: 'CHW Bettys Area',
+        parent: userContactGrandparent,
+      },
+    };
+    const conversations =  [
+      { key: 'a',
+        message: { inAllMessages: true },
+        lineage : [ 'Amy Johnsons Household', 'St Elmos Concession', 'Chattanooga Village', 'CHW Bettys Area', null]
+      },
+      { key: 'b',
+        message: { inAllMessages: true },
+        lineage : [ 'Amy Johnsons Household', 'St Elmos Concession', 'Chattanooga Village']
+      },
+      { key: 'c',
+        message: { inAllMessages: true },
+        lineage : [ 'Amy Johnsons Household', 'St Elmos Concession', 'Chattanooga Village', 'Ramdom Place', null, null]
+      },
+      { key: 'd',
+        message: { inAllMessages: true },
+        lineage : []
+      },
+      { key: 'e',
+        message: { inAllMessages: true },
+      },
+    ];
 
-    component.ngOnDestroy();
+    it('it should retrieve the hierarchy level of the connected user', async () => {
+      expect(await component.currentLevel).to.equal('parent');
+    });
 
-    expect(spySubscriptionsUnsubscribe.callCount).to.equal(1);
+    it('should not change the conversations lineage if the connected user is online only', fakeAsync(() => {
+      sinon.resetHistory();
+
+      messageContactService.getList.resolves(conversations);
+      userContactService.get.resolves(bettyOfflineUserContactDoc);
+      sessionService.isOnlineOnly.returns(true);
+
+      component.ngOnInit();
+      tick();
+      component.updateConversations({merge : true});
+      tick();
+
+      expect(component.conversations).to.deep.equal(conversations);
+    }));
+
+    it('should not change the conversations lineage if the connected user is online only', fakeAsync(() => {
```

Comment:
Is this test just exactly the same as the one before?

Reply:
yeah thanks for noticing it, it's fixed.